pkg/brokerapi/fake/fake.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fake
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/brokerapi"
+	uuid "github.com/satori/go.uuid"
+)
+
+// Client implements a fake broker API client, useful for unit testing. None of the methods on
+// the client are concurrency-safe
+type Client struct {
+	CatalogClient
+	InstanceClient
+	BindingClient
```

Comment:
>I guess this might depend on the interface that we add into which this hooks, right?

Not really sure what you mean here. `Client` is just composed of these types. The sum of all the composition means that `Client` satisfies the `BrokerClient` interface. We could build this in-code differently if we wanted, as long as it satisfies that interface. Not sure if I answered your question?